tests/integration/scripts/test_validate_integration.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Integration tests for Cylc Validate CLI script."""
+
+from cylc.flow.scripts.validate import wrapped_main as validate
+from cylc.flow.parsec.exceptions import IllegalItemError
+import pytest
+from cylc.flow.parsec.exceptions import Jinja2Error
+
+
+async def test_revalidate_checks_source(
+    _source_workflow, capsys, _setup_validate_cli
+):
+    """Validation fails if revalidating with broken config.
+    """
+    wf = _source_workflow()
+
+    setup = _setup_validate_cli({'against_source': True})
+
+    # Check that the original installation validates OK:
+    await validate(setup.parser, setup.opts, wf.opts.workflow_name)
```

Comment:
E.G. this could become (DON'T APPLY):

```suggestion
from cylc.flow.parsec.exceptions import IllegalItemError
import pytest
from cylc.flow.parsec.exceptions import Jinja2Error


async def test_revalidate_checks_source(
    _source_workflow, capsys, _setup_validate_cli
):
    """Validation fails if revalidating with broken config.
    """
    wf = _source_workflow()
    await validate(fs.path)
```